### PR TITLE
Nar'sian Beautification - Cult Constructs Resprites & Fixes

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/46431
- https://github.com/tgstation/tgstation/pull/46915 (Just the new construct)
- https://github.com/tgstation/tgstation/pull/50620
- https://github.com/tgstation/tgstation/pull/56758 
- https://github.com/tgstation/tgstation/pull/57027
- https://github.com/tgstation/tgstation/pull/59675
- https://github.com/tgstation/tgstation/pull/60680
- https://github.com/tgstation/tgstation/pull/63231

## About The Pull Request

Ports a few lovely resprites of constructs and Nar'sie from /vg/.

Included a few structural changes to Constructs, as well as a few fixes that should relieve both players and admins alike.

Included some wizard code if we ever want to give Wizard their own constructs, its commented out because I dont want this to be a balance pr. (Though Balance and Wizard are mutually exclusive)

Brought a construct over from TG. It has higher speed, but has a reduced arsenal compared to other constructs. It will only spawn during Nar'sies arrival. 

Stuff like the forges use Radial menus, so the user can see what they're getting better.

Constructs build animations look nicer.

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/175def63-8938-47e8-9b39-0fe7979c9831

Freshened up Nar'Sie all pretty.

Old:

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/d4c591e4-7e23-43f2-a9c9-f7ae7c5a1b4e

New:

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/68208edd-5727-4010-8738-9b607908b4dd

Fixes include:
- Soulstones have an extra check that should mitigate instances where spaghetti would send players to the lobby
- AI shells wont be converted instantly upon contact with the Twisted Construction spell(the check had a cyborg parameter, but didnt check if it was an occupied shell)
- The offer rune had some really buggy interactions with Cyborgs. Spaghetti code meant cyborgs could end up not being properly transferred to a soulstone and getting round removed. This is fixed.
- Dead cyborg's ghosts are pulled back into their body if they get converted to constructs. This should fix bugs with ghosts not getting properly transferred to their new bodies, or the new construct just being catatonic for the rest of the shift.

## Why It's Good For The Game

Our true god should look as pretty as possible...

Constructs have been kinda buggy for a few years, and its always irked me. This should address several of the glaring issues in their average use. 

Cyborgs getting converted, but not getting put back in their body, so now you just got this soulless husk and a round-removed player? That sucks.

Oh yeah, new sprites. They look cool.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Forge

forge

Archives

Archives

Wraith

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/d374f174-9fc9-4238-86ef-8d1be80e59e0

</details>

## Changelog
:cl:  rkz, JohnFulpWillard, RandomGamer123, IndieanaJones, itseasytosee, Fikou, actioninja, /vg/ spriters(DeityLink!), ArcaneMusic, AnturK
rscadd: New makeup & sprites on Nar'sie. Tell her how fashionable she looks next time she ~~invades~~ liberates your local plane of existence!
rscadd: New sprites for constructs. Dismember heathens, without the SOVL!
refactor: minor refactor of cult and holy versions of constructs, no ingame changes.
fix: Soulstones have an extra check that should mitigate instances where spaghetti would send players to the lobby
fix: fixes AI shells getting erroneously instantly converted by Twisted Construction spell
fix: fixes Cyborgs getting round removed by the Offer rune. They will properly transfer.
fix: fixes dead Cyborgs that are converted from being catatonic for the rest of the shift, ghosts will be yanked back inside during conversion
add: makes all cult construction gear (forge, etcetera) use radial menus, instead of lists
add: new minor construct, the Proteon! This is a weaker, but more agile construct that will be brought forth from the veil when Nar'Sie arrives, to hunt down the errant subjects of her new dominion!
/:cl:

### CODE REVIEW

**Comment by llol111** [comment]
I like the new animation for narnar, but im not a huge fan of the wierd pixel smoothing look hes got in the new version

**Review by PestoVerde322 — APPROVED** [review]
Iconwise looks good, Approved. (if only someone would give the same treatment for ratrat)

**Comment by Tsar-Salat** [comment]
> but im not a huge fan of the wierd pixel smoothing look hes got in the new version

What the hell is pixel smoothing? She has more pixels, no effect is being applied

image

**Comment by llol111** [comment]
>What the hell is pixel smoothing?
https://en.wikipedia.org/wiki/Pixel-art_scaling_algorithms

If thats not whats going on here then thats on me, it just looked like it to me.
The point is, i prefer the look with less pixels, slightly harsher corners.
https://i.imgur.com/KD5fiCB.png vs https://i.imgur.com/DjjglBD.png
just a personal preference in art styles.

Either way not something to riot about, the animation is definitely nicer in the new version.

File: code/modules/antagonists/wizard/equipment/soulstone.dm [hunk 1]
```diff
@@ -9,11 +9,15 @@
...
 	var/old_shard = FALSE
 	var/spent = FALSE
-	var/purified = FALSE
+	///This controls the color of the soulstone as well as restrictions for who can use it. THEME_CULT is red and is the default of cultist THEME_WIZARD is purple and is the default of wizard and THEME_HOLY is for purified soul stone
+	var/theme = THEME_CULT
+	/// Role check, if any needed
+	var/required_role = /datum/antagonist/cult
+
+/obj/item/soulstone/proc/role_check(mob/who)
+	return required_role ? (who.mind && who.mind.has_antag_datum(required_role, TRUE)) : TRUE
```
Comment (PowerfulBacon) [inline code/modules/antagonists/wizard/equipment/soulstone.dm:20]: Can you test this on wizards to ensure that the wizard can still use their spell "Six Soul Stone Shards and the spell Artificer"

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

**Comment by Tsar-Salat** [comment]
I changed my mind, Wizards can have their own theme.

File: code/modules/antagonists/wizard/equipment/soulstone.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/antagonists/wizard/equipment/soulstone.dm:20]: Fixed in 202c989

For the regular soulstone, the required role is set to check for cultist antag datum. 

For the soulstone/mystic subtype, the required role is redefined to check for wizard antag datum.

**Comment by DeityLink** [comment]
Hi, for credits'sake, I'm the /vg/ coder/spriter who made all the new construct sprites for Artificer/Wraith/Juggernaut (back in 2018 https://github.com/vgstation-coders/vgstation13/pull/20192) as well as the higher res Nar-Sie sprite (back in... holy shit almost a decade ago https://github.com/vgstation-coders/vgstation13/pull/1711).

**Comment by Tsar-Salat** [comment]
> holy shit almost a decade ago

wow you're old /s

The old sprites we have look like ass, thanks for making these ones, they look beautiful. I'll make sure to give you credit, thanks for contacting me!

**Comment by Tsar-Salat** [comment]
hiii can I get a tm or somethin

**Review by PestoVerde322 — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**